src/backend/access/external/pxfutils.c

```diff
@@ -47,6 +52,24 @@ void port_to_str(char **port, int new_port)
 }
 
 /*
+ * get_hdfs_location_from_filespace
+ *
+ * Get hdfs location from pg_filespace_entry
+ * The returned path needs to be pfreed by the caller.
+ */
+void get_hdfs_location_from_filespace(char** path)
+{
+	Assert(NULL != path);
```

Comment:
If I remember correctly a table can specify a table space that is different from the table space used by the database. Will it cause trouble with this code?